### PR TITLE
aws.glue-catalog | cross-account filter

### PR DESCRIPTION
**Pull request by PratMis**
This PR adds supports for cross-account filter for glue-catalog resource

Gets resource policy using get_resource_policy() and uses `CrossAccountAccessFilter` to evaluate the policy

### CODE REVIEW

File: c7n/resources/glue.py [hunk 1]
```diff
@@ -516,3 +516,37 @@ class GlueCatalogEncryptionFilter(GlueCatalogEncryptionEnabled):
...
+            return r[self.policy_annotation]
+        try:
+            policy = client.get_resource_policy().get('PolicyInJson')
+        except client.exceptions.EntityNotFoundException:
+            policy = {}
+        r[self.policy_annotation] = policy
+        return policy
+
+    def __call__(self, r):
+        if self.get_resource_policy(r):
```
Comment (kapilt) [inline c7n/resources/glue.py:551]: this is pretty broken on logic

Comment (PratMis) [reply c7n/resources/glue.py:551]: yep, agree with you @kapilt! I should have given it a thought myself. Switching it back to our logic in `CrossAccountAccessFiter`. I think that'll make it a lot better. Thanks

File: tests/data/placebo/test_glue_catalog_cross_account/glue.GetResourcePolicy_1.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+    "status_code": 200,
+    "data": {
+        "PolicyInJson": "{\n  \"Version\" : \"2012-10-17\",\n  \"Statement\" : [ {\n    \"Effect\" : \"Allow\",\n    \"Principal\" : {\n      \"AWS\" : \"arn:aws:iam::123123456789:root\"\n    },\n    \"Action\" : \"glue:CreateTable\",\n    \"Resource\" : \"arn:aws:glue:us-east-1:644160558196:catalog\"\n  } ]\n}",
```
Comment (PratMis) [inline tests/data/placebo/test_glue_catalog_cross_account/glue.GetResourcePolicy_1.json:4]: The account number in the Principal would get overridden to our default account here. I changed it to `123123456789` which originally was something else(external AWS account) to support the test case. Let me know if that makes sense to you!! Thanks

Comment (kapilt) [reply tests/data/placebo/test_glue_catalog_cross_account/glue.GetResourcePolicy_1.json:4]: yeah.. we need a flag to disable the account rewrite stuff

**Pull request merged**